ndk/src/events/nip13/nip13.ts

```diff
@@ -0,0 +1,88 @@
+import { serializeEvent } from "nostr-tools";
+import type { UnsignedEvent } from "nostr-tools";
+import type { NDKEvent } from "../index.js";
+import Worker from './pow.worker.js';
+import { deserializeEvent, getNonceBounds } from "./Miner";
+
+let worker: Worker;
+
+export async function pow(this: NDKEvent, target: number) {
+  if (!this.ndk) throw new Error("No NDK instance found!");
+
+  const intTarget = Math.floor(target);
+
+  // check if event has necessary props:
+  if (!this.kind) throw new Error("NIP-13: Event has no kind");
+  if (!this.created_at) throw new Error("NIP-13: Event has no created_at");
+  if (!this.pubkey) throw new Error("NIP-13: Event has no pubkey");
+  if (!this.content) console.warn("NIP-13: Event has no content");
+  if (intTarget !== target) console.warn("NIP-13: Target is not an integer")
+
+  // can't do POW if these things are already present
+  if (this.id) throw new Error("NIP-13: Event already has an id");
+  if (this.sig) throw new Error("NIP-13: Event already has a signature");
+
+  // spawn worker
+  if (!worker) {
+    worker = new Worker();
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-this-alias
+  const _event = this;
+
+  return new Promise((resolve, reject) => {
+
+    // declare handler for worker messages in closure
+    function handleMinedPOW(msg: MessageEvent) {
+      if (msg.data.status === 'pow-target-found') {
+        worker.postMessage({command: 'stop'}); // not strictly necessary as the worker stops on its own when the target is found
+        const binary = msg.data.binary;
+        const eventSerialized = new TextDecoder().decode(binary);
+        const raw = deserializeEvent(eventSerialized);
```

Comment:
Do you need your own `deserialize` event here? Why not just use `NDKEvent.rawEvent()`?

Reply:
Nope! Thanks for pointing that out! I will switch that.

Reply:
`NDKEvent.rawEvent()` would work if the event was already an `NDKEvent`, but here we are going from binary ➡ serialized ➡ deserialized in order to extract the nonce created by the miner.

Is there a way to pass a binary or serialized event into NDK to convert it into an `NDKEvent` and then call `.rawEvent()`?